mirrord/layer/src/file/ops.rs

```diff
@@ -263,7 +273,7 @@ pub(crate) fn fdopendir(fd: RawFd) -> Detour<usize> {
         .insert(local_dir_fd as usize, remote_dir_fd);
 
     // According to docs, when using fdopendir, the fd is now managed by OS - i.e closed
-    open_files.remove(&fd);
+    OPEN_FILES.lock()?.remove(&fd);
```

Comment:
Lock (and free) here again, instead of holding locked all the way from the top down here.